src/adr/0034-distributed-listeners-configuration.md

```diff
@@ -0,0 +1,106 @@
+user implemented mailbox listeners is partially deployed.
+
+This differs from [26. Removing a configured additional MailboxListener](0026-removing-configured-additional-mailboxListeners.md)
+by the fact we no longer need to register all listeners at once.
+
+A WebAdmin endpoint will allow:
+
+ - **to add a listener** to the one configured. Such a call:
+    - Will fail if the listener is unknown to James, or if the corresponding group is unknown.
+    - Upon success the listener is added to the **configured mailbox listener aggregate**, and the listener is 
+    registered locally. No broadcast is attempted, meaning that other James servers will need a reboot to actually start 
+    consuming the queue.
+ - **to remove a listener**. Such a call:
+    - Will fail if the listener is required by Guice bindings on the current server (distributed check will not be 
+    implemented) or if the listener is not configured.
+    - Upon success, the listener is removed from the **configured mailbox listener aggregate**, and the listener is 
+    unregistered locally. No broadcast is attempted, meaning that other James servers will need a reboot to actually stop 
+    consuming the queue. However, new events will stop arriving in the queue as its binding will be removed.
+
+Integration tests relying on additional mailbox listeners of the distributed James product will require to be ported to 
+perform additional mailbox listener registry with this WebAdmin endpoint. JMAP SpamAssassin, quota mailing tests are 
+concerned.
+
+We will also expose an endpoint listing the groups currently in use, and for each group the associated configuration, if 
+any. This will query the **configured mailbox listener aggregate**.
+
+We will introduce a health check to actually ensure that RabbitMQ resources match the configured listeners, and propose
+a WebAdmin endpoint to add/remove bindings/queue in a similar fashion of what had been proposed in 
+[26. Removing a configured additional MailboxListener](0026-removing-configured-additional-mailboxListeners.md).
+
+## Consequences
+
+All products other than "Distributed James" are unchanged.
+
+All the currently configured additional listeners will need to be registered.
+
+The definition of mailbox listeners is thus centralized and we are not exposed to an heterogeneous configuration 
+incident.
+
+## Possible evolutions
+
+A broadcast can be attempted to propagate eventBus topology changes:
+
+ - Each James server registers an exclusive queue to an "eventBus topology change" exchange.
+ - Upon modification of the actual topology an "add" or "remove" event is emitted.
+ - Each running James reacts to these events by instantiating the corresponding listener and starting consuming the 
+ associated queue, or stops consuming the associated queue.
+ 
+If a listener is added but is not in the classpath, an ERROR log is emitted. This can happen during a rolling upgrade,
```

Comment:
how about passing the listener jar as the endpoint payload and put it into the blobstore?

Reply:
That's clearly overkill to me !

Let's solve one problem at a time, distributed JAR registry is only loosely coupled with that problematic.